pkg/provider/kind/kind.go

```diff
@@ -0,0 +1,209 @@
+	if err := c.checkDeploymentVarsAndFiles(); err != nil {
+		return err
+	}
+
+	deploymentResource, err := provider.DeploymentsParse(c.DeploymentFiles, c.DeploymentVars)
+	if err != nil {
+		log.Fatalf("Couldn't parse deployment files: %v", err)
+	}
+	for _, deployment := range deploymentResource {
+
+		decode := scheme.Codecs.UniversalDeserializer().Decode
+		k8sObjects := make([]runtime.Object, 0)
+
+		for _, text := range strings.Split(string(deployment.Content), provider.Separator) {
+			text = strings.TrimSpace(text)
+			if len(text) == 0 {
+				continue
+			}
+
+			resource, _, err := decode([]byte(text), nil, nil)
+			if err != nil {
+				return errors.Wrapf(err, "decoding the resource file:%v, section:%v...", deployment.FileName, text[:100])
+			}
+			if resource == nil {
+				continue
+			}
+			k8sObjects = append(k8sObjects, resource)
+		}
+		if len(k8sObjects) > 0 {
+			c.k8sResources = append(c.k8sResources, k8sProvider.Resource{FileName: deployment.FileName, Objects: k8sObjects})
+		}
+	}
+	return nil
+}
+
+// checkDeploymentVarsAndFiles checks whether the requied deployment vars are passed.
+func (c *KIND) checkDeploymentVarsAndFiles() error {
+	reqDepVars := []string{"PR_NUMBER", "CLUSTER_NAME"}
+	for _, k := range reqDepVars {
+		if v, ok := c.DeploymentVars[k]; !ok || v == "" {
+			return fmt.Errorf("missing required %v variable", k)
+		}
+	}
+	if len(c.DeploymentFiles) == 0 {
+		return fmt.Errorf("missing deployment file(s)")
+	}
+	return nil
+}
+
+// ClusterCreate create a new cluster or applies changes to an existing cluster.
+func (c *KIND) ClusterCreate(*kingpin.ParseContext) error {
+	for _, deployment := range c.kindResources {
+		CreateWithConfigFile := cluster.CreateWithRawConfig(deployment.Content)
+
+		err := c.kindProvider.Create(c.DeploymentVars["CLUSTER_NAME"], CreateWithConfigFile)
+		if err != nil {
+			log.Fatalf("creating cluster err:%v", err)
+		}
+	}
+	return nil
+}
+
+// ClusterDelete deletes a k8s cluster.
+func (c *KIND) ClusterDelete(*kingpin.ParseContext) error {
+	err := c.kindProvider.Delete(c.DeploymentVars["CLUSTER_NAME"], homedir.HomeDir()+"/.kube/config")
+	if err != nil {
+		log.Fatalf("creating cluster err:%v", err)
+	}
+	return nil
+}
+
+// NewK8sProvider sets the k8s provider used for deploying k8s manifests.
+func (c *KIND) NewK8sProvider(*kingpin.ParseContext) error {
+	var err error
+	apiConfig, err := clientcmd.LoadFromFile(homedir.HomeDir() + "/.kube/config")
+	if err != nil {
+		log.Fatal("failed to load user provided kubeconfig", err)
+	}
+
+	c.k8sProvider, err = k8sProvider.New(c.ctx, apiConfig)
+	if err != nil {
+		log.Fatal("k8s provider error", err)
+	}
+	return nil
+}
+
+// ResourceApply calls k8s.ResourceApply to apply the k8s objects in the manifest files.
+func (c *KIND) ResourceApply(*kingpin.ParseContext) error {
+	if err := c.k8sProvider.ResourceApply(c.k8sResources); err != nil {
+		log.Fatal("error while applying a resource err:", err)
```

Comment:
It'll be nice to have all `log.Fatal` replaced by returning the error in this file if possible :)